alerts/AggregatedAPIErrors.md

```diff
@@ -0,0 +1,65 @@
+v1.apps.openshift.io                            openshift-apiserver/api                                      True        340d
+v1.authorization.openshift.io                   openshift-apiserver/api                                      True        340d
+v1.build.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.image.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.oauth.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.packages.operators.coreos.com                openshift-operator-lifecycle-manager/packageserver-service   True        340d
+v1.project.openshift.io                         openshift-apiserver/api                                      True        340d
+v1.quota.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.route.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.security.openshift.io                        openshift-apiserver/api                                      True        340d
+v1.template.openshift.io                        openshift-apiserver/api                                      True        340d
+v1.user.openshift.io                            openshift-apiserver/api                                      True        340d
+v1beta1.metrics.k8s.io                          openshift-monitoring/prometheus-adapter                      True        340d
+```
+Here we want the availability status to be `True`. A `False` means that requests for that API service, API server pods, or resources belonging to that apiGroup failed many times during the last minutes.
+
+##### Check the `apiserver` pods
+```shell
+$ oc get pods -n openshift-apiserver
+NAME                        READY   STATUS    RESTARTS   AGE
+apiserver-c74d6bd79-pkmw4   1/1     Running   0          41d
+apiserver-c74d6bd79-t4l9d   1/1     Running   0          41d
+apiserver-c74d6bd79-z6qxx   1/1     Running   0          41d
+```
+If their status is not `Running` check the logs for more details. 
+
+##### Check the authentication certificates of the aggregated API
+Make sure the certificates are up to date and still valid. Use:
+```shell
+$ oc get configmaps -n kube-system extension-apiserver-authentication
+```
+you can save those certificates into a file and use the following command to check the end dates:
+```shell
+$ openssl x509 -noout -enddate -in {myfile_with_certs.crt}
+```
+Those certificates are used by the aggregated APIs to validate requests.
+
+## Mitigation
+In case a certificate expiry can be excluded but you observed some hanging pods, try to find issues in the logs. However, as these pods are controlled by a deployment, you can restart them when they are not answering requests.
```

Comment:
Mitigation seems like it deserves separate sections under each diagnosis entry?  Really, this whole thing seems like it should be a directed-graph flowchart, and trying to flatten it into a strict header tree seems like it will be difficult without lots of internal cross-links, especially if we try to take all of the terminal-nodes from that flow-chart and collect them as sibling subsections here.  Can we drop *Diagnosis*, and restructure like:

> ## Mitigation
>
> This alert may fire in many situations, and appropriate mitigation depends on the underlying issue.  Diagnose and mitigate by following the following procedure.
>
> ### Check the APIServices
>
> ```console
> $ oc get apiservices
> ...
> ```
>
> If any are `False`, proceed to [debugging failing APIServices](#debugging-failing-apiservices).  If all are `True`, proceed to [check the authentication certificates of the aggregated API](#check-the-authentication-certificated-of-the-aggregated-api).
>
> ### Debugging failing APIServices
> ...
> ### Check the authentication certificates of the aggregated API
> ...

Reply:
I stumbled upon that as well. I wanted to stick to the standardized structure for all costs. 
But I also agree with your proposed structure. Maybe we can find still something for a very generic initial diagnosis?
Can we read something from the alert information? Otherwise, I would move the subheaders to Mitigation.
`- alertname = AggregatedAPIErrors
 - name = v1.packages.operators.coreos.com
 - namespace = default
 - prometheus = openshift-monitoring/k8s
 - severity = warning
Annotations:
 - message = An aggregated API v1.packages.operators.coreos.com/default has reported errors. The number of errors have increased for it in the past five minutes. High values indicate that the availability of the service changes too often.
 `

Reply:
I hope that we don't have all that many cases where the diagnostic steps and the mitigation steps are so involved that they need their own multi-part sections.  And ideally we don't have many alerts that feed into a multi-option diagnostic tree.  Or maybe the fact that the alert is firing is _already_ enough to diagnose the issue?  So for those well-scoped alerts, it would just be:

```markdown
# WidgetStamperDisabled

## Impact

A disabled widget stamper will starve the cluster of widgets.  OpenShift updates consume widgets, so updates will fail as long as the widget stamper is disabled.

## Mitigation

Turn the widget stamper back on with `oc ...`.
```

For more alerts with a broader exposure like `AggregatedAPIErrors`, I think the single *Mitigation* section is still a good fit.  You've already covered the impact in the *Impact* section; folks who are still reading are interested in mitigation.  Maybe you need to flow them through a decision tree before you suggest a mitigation command.  Maybe not.  But either way, I don't see an upside to a separate section separating the optional decision tree from the mitigation steps at the end of that tree.